core/modules/ccontrol/UserQuerySelect.cc

```diff
@@ -272,25 +287,17 @@ void UserQuery::discard() {
         // Silence merger discarding errors, because this object is being released.
         // client no longer cares about merger errors.
     }
-    LOGF_INFO("Discarded UserQuery(%1%)" % _sessionId);
-}
-
-/// Constructor. Most setup work done by the UserQueryFactory
-UserQuery::UserQuery(std::shared_ptr<qproc::QuerySession> qs, qmeta::CzarId czarId)
-    :  _messageStore(std::make_shared<qdisp::MessageStore>()),
-       _qSession(qs), _qMetaCzarId(czarId), _qMetaQueryId(0),
-       _killed(false), _submitted(false), _sessionId(0), _sequence(0) {
-    // Some configuration done by factory: See UserQueryFactory
+    LOGF_INFO("Discarded UserQuerySelect(%1%)" % _sessionId);
 }
 
 /// Setup merger (for results handling and aggregation)
-void UserQuery::_setupMerger() {
+void UserQuerySelect::_setupMerger() {
     LOG(getLogger(), LOG_LVL_TRACE, "Setup merger");
     _infileMergerConfig->mergeStmt = _qSession->getMergeStmt();
     _infileMerger = std::make_shared<rproc::InfileMerger>(*_infileMergerConfig);
 }
 
-void UserQuery::_setupChunking() {
+void UserQuerySelect::setupChunking() {
```

Comment:
Yes, this method is called by UserQueryFactory. UserQueryFactory was a friend (:yuck:) of UserQuery class so it could call private methods. I killed that friendship and updated UserQuerySelect interface to make this method public.